src/UI/Table.elm

```diff
@@ -26,27 +28,45 @@ import UI.Button as Button exposing (Button)
 import UI.Internal.Basics exposing (..)
 import UI.Internal.NList as NList exposing (NList)
 import UI.Internal.Palette as Palette
+import UI.Internal.ToggableList as ToggableList
 import UI.Internal.TypeNumbers as T
 import UI.Palette as Palette exposing (brightnessMiddle, tonePrimary)
-import UI.RenderConfig exposing (RenderConfig)
+import UI.RenderConfig as RenderConfig exposing (RenderConfig)
 import UI.Text as Text exposing (Text)
+import UI.Utils.Element exposing (zeroPadding)
 
 
-type Table msg columns
-    = Table (Properties columns) (Options msg columns)
+type Table msg object columns
+    = Table (Properties columns) (Options msg object columns)
 
 
 type alias Properties columns =
     { headers : HeaderRow columns
     }
 
 
-type alias Options msg columns =
-    { rows : List (Row msg columns)
+type alias Options msg object columns =
+    { rows : Rows msg object columns
     , width : Element.Length
     }
 
 
+type alias ResponsiveOptions msg object columns =
+    { detailsShowLabel : String
+    , detailsCollapseLabel : String
+    , selectMsg : object -> msg
+    , isSelected : object -> Bool
+    , coverView : RenderConfig -> Palette.Color -> object -> Bool -> Element msg
+    , toRow : object -> Row msg columns
```

Comment:
Ok so thaaat's is the complicated part, if we look at the Events page, we have a duplicated info:

![Screenshot 2020-06-17 at 13 12 35](https://user-images.githubusercontent.com/1559013/84891577-64c15500-b09c-11ea-982b-bff4a2bbf3d1.png)

I think in Zeplin the `created at` does not appear in the details, is something like, we would need to choose what fields we want to show in the detailed version 🤔